cmake/CompilerFlags.cmake

```diff
@@ -234,7 +234,10 @@ if("Ninja" STREQUAL ${CMAKE_GENERATOR})
   include(CheckCXXCompilerFlag)
   # Clang
   if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-    AddFlagIfSupported(-fcolor-diagnostics COMPILER_SUPPORTS_fdiagnostics_color)
+    # On Mac with Ninja (kitware binary for fortran support) and brew gfortran, I get build errors due to this flag.
+    if (NOT BUILD_FORTRAN)
+      AddFlagIfSupported(-fcolor-diagnostics COMPILER_SUPPORTS_fdiagnostics_color)
+    endif()
   endif()
```

Comment:
Interesting, I've never had any issues.  Is there any chance we should protect this more specifically for your corner case so that it still works OK for other builds?

Reply:
I've never had any issues with it either until today, and I **think** I had built fortran on mac with ninja before. I don't think it matters that much if no one bumps into this anyways, but there should be a way here to use `add_compile_options` with generator expressions to limit to everything BUT fortran instead of disabling for everyone.

Reply:
Yeah `add_compile_options($<$<NOT:$<COMPILE_LANGUAGE:Fortran>>:${flag}>)` works